ArduPlane/quadplane.cpp

```diff
@@ -810,22 +824,53 @@ void QuadPlane::multicopter_attitude_rate_update(float yaw_rate_cds)
 {
     check_attitude_relax();
 
-    // tailsitter-only bodyframe roll control options
-    if (is_tailsitter()) {
+    // tailsitter-only body-frame roll control options
+    // Angle mode attitude control for pitch and body-frame roll, rate control for euler yaw.
+    if (is_tailsitter() &&
+        ((tailsitter.input_type == TAILSITTER_INPUT_BF_ROLL_M) ||
+         (tailsitter.input_type == TAILSITTER_INPUT_BF_ROLL_P))) {
+
+        // Since function input_euler_rate_yaw_euler_angle_pitch_bf_roll()
+        // needs body-frame roll and yaw inputs, it is necessary to rescale
+        // the plane inputs nav_roll_cd and yaw_rate_cds to represent
+        // yaw rate and roll angle, respectively.
+
+        // Get the roll and yaw ranges from parameters
+        int16_t roll_limit = plane.quadplane.aparm.angle_max;
+        // separate limit for tailsitter roll, if set
+        if (plane.quadplane.tailsitter.max_roll_angle > 0) {
+            roll_limit = plane.quadplane.tailsitter.max_roll_angle * 100.0f;
+        }
+
+        // Constrain the roll angle limit set by parameters
+        roll_limit = constrain_int16(roll_limit, -9000, 9000);
+
+        // Guard against a bad parameter value
+        float yaw_rate_limit = constrain_float(plane.quadplane.yaw_rate_max, 20, 500) * 100.0f;
```

Comment:
I don't thing we should contrain the param, maybe just check that its not zero.

Reply:
I was worried that the body-frame roll code might do unexpected things with a very large yaw rate, and I think a max of 20 deg/sec is quite slow. I was also looking at @Range: 50 500 in the metadata for YAW_RATE_MAX; I suppose those tend to be just suggestions, but it's hard for a plane to stay airborne while yawing at high speed. I find in SITL that I lose all airspeed and stall at less than 120 deg/sec.

Reply:
I would just put them as the recommended range in the wiki, we let people set loads of parameters to values that will crash there vehicles. Then the people who do need the extra range can still have it.

Reply:
I guess we could just constrain it to be > 0 then, I wonder if there are any sanity checks in the attitude control code? It doesn't make sense to try to fly with bad parameters there.

Reply:
relaxed the constraint to require non-zero max yaw rate (min .01 deg/sec). It must be non-zero in order for  tailsit_input type 3 (plane-mode body-frame roll) to work correctly.